idaes/models/properties/examples/tests/test_methanol_reactions.py

```diff
@@ -0,0 +1,187 @@
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Author: Andrew Lee
+"""
+import pytest
+from pyomo.environ import (check_optimal_termination,
+                           Block,
+                           ConcreteModel,
+                           Set,
+                           value)
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    fixed_variables_set,
+    activated_constraints_set,
+)
+from idaes.core.solvers import get_solver
+
+from idaes.models.properties.modular_properties.base.generic_property import (
+    GenericParameterBlock,
+)
+from idaes.models.properties.modular_properties.base.generic_reaction import (
+    GenericReactionParameterBlock,
+)
+
+from idaes.models.properties.examples.methanol_ideal_VLE import \
+    config_dict as vap_config_dict
+from idaes.models.properties.examples.methanol_reactions import \
+    config_dict as rxn_config_dict
+
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestParamBlock(object):
+    @pytest.mark.unit
+    def test_build(self):
+        model = ConcreteModel()
+        model.thermo_params = GenericParameterBlock(default=vap_config_dict)
+
+        model.rxn_params = GenericReactionParameterBlock(
+            default={"property_package": model.thermo_params,
+                     **rxn_config_dict}
+        )
+
+        rate_config = model.rxn_params.config.rate_reactions
+
+        assert isinstance(model.rxn_params.rate_reaction_idx, Set)
+        assert len(model.rxn_params.rate_reaction_idx) == 1
+        assert "R1" in model.rxn_params.rate_reaction_idx
+
+        assert isinstance(model.rxn_params.rate_reaction_stoichiometry, dict)
+        assert len(model.rxn_params.rate_reaction_stoichiometry) == 5
+        for k, v in model.rxn_params.rate_reaction_stoichiometry.items():
+            if (k[1], k[2]) in rate_config[k[0]].stoichiometry.keys():
+                assert v == rate_config[k[0]].stoichiometry[k[1], k[2]]
+            else:
+                assert v == 0
+
+        assert isinstance(model.rxn_params.reaction_R1, Block)
+
+        assert_units_consistent(model)
+
+
+class TestStateBlock(object):
```

Comment:
Similar to the thermo properties, I think most of these tests can be dropped - all you need to test here is that you get the expected result.